src/eventsub/channel/moderate.rs

```diff
@@ -0,0 +1,1293 @@
+        moderator_user_id: impl Into<types::UserId>,
+    ) -> Self {
+        Self {
+            broadcaster_user_id: broadcaster_user_id.into(),
+            moderator_user_id: moderator_user_id.into(),
+        }
+    }
+}
+
+impl EventSubscription for ChannelModerateV1 {
+    type Payload = ChannelModerateV1Payload;
+
+    const EVENT_TYPE: EventType = EventType::ChannelModerate;
+    #[cfg(feature = "twitch_oauth2")]
+    const SCOPE: twitch_oauth2::Validator = twitch_oauth2::validator![all(
+        any(
+            twitch_oauth2::Scope::ModeratorReadBlockedTerms,
+            twitch_oauth2::Scope::ModeratorManageBlockedTerms
+        ),
+        any(
+            twitch_oauth2::Scope::ModeratorReadChatSettings,
+            twitch_oauth2::Scope::ModeratorManageChatSettings
+        ),
+        any(
+            twitch_oauth2::Scope::ModeratorReadUnbanRequests,
+            twitch_oauth2::Scope::ModeratorManageUnbanRequests
+        ),
+        any(
+            twitch_oauth2::Scope::ModeratorReadBannedUsers,
+            twitch_oauth2::Scope::ModeratorManageBannedUsers
+        ),
+        any(
+            twitch_oauth2::Scope::ModeratorReadChatMessages,
+            twitch_oauth2::Scope::ModeratorManageChatMessages
+        ),
+        twitch_oauth2::Scope::ModeratorReadModerators,
+        twitch_oauth2::Scope::ModeratorReadVips,
+    )];
+    const VERSION: &'static str = "1";
+}
+
+// XXX: this struct can never be deny_unknown_fields as it has flattened fields
+/// [`channel.moderate`](ChannelModerateV1) response payload.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct ChannelModerateV1Payload {
+    /// The ID of the broadcaster.
+    pub broadcaster_user_id: types::UserId,
+    /// The login of the broadcaster.
+    pub broadcaster_user_login: types::UserName,
+    /// The user name of the broadcaster.
+    pub broadcaster_user_name: types::DisplayName,
+
+    /// The channel in which the action originally occurred. Is the same as the `broadcaster_user_id` if not in shared chat.
+    pub source_broadcaster_user_id: Option<types::UserId>,
+    /// The channel in which the action originally occurred. Is the same as the `broadcaster_user_login` if not in shared chat.
+    pub source_broadcaster_user_login: Option<types::UserName>,
+    /// The channel in which the action originally occurred.
+    ///
+    /// Is [None] when the moderator action happens in the same channel as the broadcaster.
+    /// Is [Some] when in a shared chat session, and the action happens in the channel of a participant other than the broadcaster.
+    pub source_broadcaster_user_name: Option<types::DisplayName>,
+
+    /// The broadcaster user ID.
+    pub moderator_user_id: types::UserId,
+    /// The broadcaster display name.
+    pub moderator_user_name: types::DisplayName,
+    /// The broadcaster login.
+    pub moderator_user_login: types::UserName,
+    /// The action being taken
+    #[serde(flatten)]
```

Comment:
This is a bit unfortunate as it means that serde will parse unknown fields into some intermediate map, which is then passed to `ActionV1::deserialize`, because it doesn't know the fields of `ActionV1`.